SDKBuildScripts/Update-PlayFabSdk.ps1

```diff
@@ -0,0 +1,293 @@
+    [ValidateSet(
+        "ActionScriptSDK",
+        "Cocos2d-xSDK",
+        "CSharpSDK",
+        "JavaSDK",
+        "JavaScriptSDK",
+        "LuaSDK",
+        "NodeSDK",
+        "Objective_C_SDK",
+        "PhpSDK",
+        "PostmanCollection",
+        "PythonSDK",
+        "SdkTestingCloudScript",
+        "UnrealMarketplacePlugin",
+        "UnitySDK",
+        "WindowsSDK",
+        "XPlatCppSDK")]
+    [string[]]$SdkNames,
+    [Parameter(ParameterSetName="ApiSpecPath", ValueFromPipelineByPropertyName = $true)]
+    [AllowEmptyString()]
+    [string]$ApiSpecPath,
+    [Parameter(ParameterSetName="ApiSpecPfUrl", ValueFromPipelineByPropertyName = $true)]
+    [AllowEmptyString()]
+    [string]$ApiSpecPfUrl,
+    [Parameter(ParameterSetName="ApiSpecPfUrlCloudVertical", ValueFromPipelineByPropertyName = $true)]
+    [string]$Vertical = "master",
+    [Parameter(ParameterSetName="ApiSpecPfUrlCloudVertical", ValueFromPipelineByPropertyName = $true)]
+    [string]$Cloud,
+    [Parameter(ParameterSetName="ApiSpecGitUrl", ValueFromPipelineByPropertyName = $true)]
+    [AllowEmptyString()]
+    [string]$ApiSpecGitUrl,
+    [Parameter(ValueFromPipelineByPropertyName = $true)]
+    [string]$OutputPath = "..\..\sdks",
+    [Parameter(ValueFromPipelineByPropertyName = $true)]
+    [string]$TargetSource,
+    [Parameter(ValueFromPipelineByPropertyName = $true)]
+    [switch]$KeepSource,
+    [Parameter(ValueFromPipelineByPropertyName = $true)]
+    [switch]$NonNullable,
+    [Parameter(ValueFromPipelineByPropertyName = $true)]
+    [switch]$Beta
+)
+
+begin
+{
+    if(!(Get-Command node))
+    {
+        throw "You must have Node.js installed to generate the SDK"
+    }
+
+    $sdkTargetSrcMap = @{
+        "ActionScriptSDK" = "actionscript";
+        "Cocos2d-xSDK" = "cpp-cocos2dx";
+        "CSharpSDK" = "csharp";
+        "JavaSDK" = "java";
+        "JavaScriptSDK" = "javascript";
+        "LuaSDK" = "LuaSdk";
+        "NodeSDK" = "js-node";
+        "Objective_C_SDK" = "objc";
+        "PhpSDK" = "PhpSdk";
+        "PostmanCollection" = "postman";
+        "PythonSDK" = "PythonSdk";
+        "SdkTestingCloudScript" = "SdkTestingCloudScript";
+        "UnrealMarketplacePlugin" = "UnrealMarketplacePlugin";
+        "UnitySDK" = "unity-v2";
+        "WindowsSDK" = "windowssdk";
+        "XPlatCppSDK" = "xplatcppsdk";
+    }
+
+    $sdksPath = $OutputPath
+    if(![System.IO.Path]::IsPathRooted($sdksPath))
+    {
+        $sdksPath = Resolve-Path (Join-Path $PSScriptRoot $OutputPath)
+    }
+
+    if(!(Test-Path $sdksPath))
+    {
+        mkdir $sdksPath | Out-Null
+    }
+
+    if($PSCmdlet.ParameterSetName -eq "ApiSpecPfUrlCloudVertical")
+    {
+        if($Vertical -or $Cloud)
+        {
+            # If either cloud or vertical is not provided, we want to remove the leading or trailing dot.
+            $cloudVertical = "$Vertical.$Cloud".TrimEnd(".")
+            $ApiSpecPfUrl = "https://$($cloudVertical).playfabapi.com/apispec"
```

Comment:
is "$($" correct here? That seems wrong.

Reply:
This is correct but not strictly necessary., `$(...)` just says evaluate what's in here and put it in the string.  Sometimes you can have issues if you're doing stuff like `"One_$Value_Two"` which will look for a variable called $Value_Two, so you have to surround it in $(...) to prevent that.

I'm removing it.

Reply:
You can see the _ scenario I talked about at the end of this file.